### PR TITLE
enhancement/optimize Speed for ConnKeys generation

### PR DESCRIPTION
**Pull request by alazarevski**
Optimized performance and UI hanging for pre-key generation

### CODE REVIEW

File: src/actions/connectionKeyPairActions.js [hunk 1]
```diff
@@ -232,19 +264,24 @@ export const updateConnectionKeyPairs = (
...
+              totalConnections,
+              connectionKeyPairs.length,
+              lastConnectionKeyIndex);
+          const resultConnectionKeys = connectionKeyPairs.concat(newKeyPairs);
+          await dispatch({
+            type: UPDATE_CONNECTION_KEY_PAIRS,
+            payload: resultConnectionKeys,
+          });
+          await dispatch(saveDbAction('connectionKeyPairs', { connectionKeyPairs: resultConnectionKeys }, true));
+          dispatch(backgroundPreKeyGeneratorAction(mnemonic, privateKey));
```
Comment (grilix) [inline src/actions/connectionKeyPairActions.js]: We could move this line outside the `if` and drop the `else` part.

Comment (alazarevski) [reply src/actions/connectionKeyPairActions.js]: I was worried if the dispatched action would affect the UI if the context of the setTimeout changed for react-native

File: src/actions/connectionKeyPairActions.js [hunk 2]
```diff
@@ -206,6 +207,37 @@ export const updateOldConnections = (oldConnectionCount: number, theWalletId?: ?
   };
 };
 
+
+export const backgroundPreKeyGeneratorAction = (mnemonic: ?string, privateKey: ?string, insideRun?: ?boolean) => {
+  return async (dispatch: Function, getState: Function) => {
+    if (insideRun) {
```
[the remark this answers is not in the document]
Comment (alazarevski) [reply src/actions/connectionKeyPairActions.js]: I was worried if setTimeout in a different redux action would affect the UI thread when the components switch. This had the optimal results after the caller has been executed

[the remark this answers is not in the document]
Comment (grilix) [reply src/actions/connectionKeyPairActions.js]: Alright, yeah, I'm not sure about that.

**Pull request merged**